benchmarker/scripts/python/generate-filtered-beir-dataset.py

```diff
@@ -0,0 +1,99 @@
+import pandas as pd
+import numpy as np
+from datasets import load_dataset
+import h5py
+import faiss, json, random
+import os
+
+NEGATIVE_FILTERS = False
+
+datasets = ['arguana', 'cqadupstack-android', 'cqadupstack-english', 'cqadupstack-gaming', 'cqadupstack-gis', 'cqadupstack-mathematica', 'cqadupstack-physics', 'cqadupstack-programmers', 'cqadupstack-stats', 'cqadupstack-text', 'cqadupstack-unix', 'cqadupstack-webmasters', 'cqadupstack-wordpress', 'fiqa', 'nfcorpus', 'quora', 'robust04', 'scidocs', 'scifact', 'trec-covid', 'trec-news', 'webis-touche2020']
+
+def process_dataset(dataset_name):
+    df_corpus = load_dataset("Cohere/beir-embed-english-v3", f"{dataset_name}-corpus", split="train")
+    embeddings = df_corpus[:50000]['emb']
+    dataset_names = [dataset_name] * len(embeddings)
+    return pd.DataFrame({'embedding': embeddings, 'dataset': dataset_names})
+
+all_data = pd.concat([process_dataset(dataset) for dataset in datasets], ignore_index=True)
+all_data['embedding'] = all_data['embedding'].apply(lambda x: np.array(x, dtype=np.float32))
+all_data = all_data.sample(frac=1, random_state=42).reset_index(drop=True)
+
+train_size = len(all_data) - 10000
+train_data = all_data[:train_size]
+test_data = all_data[train_size:]
+
+dataset_to_id = {dataset: idx for idx, dataset in enumerate(datasets)}
+
+def generate_json_properties(category_id):
+    return json.dumps({"category": str(category_id)})
+
+def get_random_different_category(original_category, num_categories):
+    categories = list(range(num_categories))
+    categories.remove(original_category)
+    return random.choice(categories)
+
+train_categories = np.array([dataset_to_id[dataset] for dataset in train_data['dataset']], dtype=np.int64)
+test_categories = np.array([dataset_to_id[dataset] for dataset in test_data['dataset']], dtype=np.int64)
+
+if NEGATIVE_FILTERS:
```

Comment:
What is the purpose of this `NEGATIVE_FILTERS`?

Reply:
Fixed missing requirements, `NEGATIVE_FILTERS` is used to invert the filters for negative query <> filter correlation.